napalm/base/base.py

```diff
@@ -218,9 +218,33 @@ def compare_config(self):
         """
         raise NotImplementedError
 
-    def commit_config(self, message=""):
+    def commit_config(self, message="", confirm_timeout=None):
```

Comment:
I think we also need a `commit_confirm=False` argument (i.e. in addition to the `confirm_timeout` we also need a second argument indicating you are doing `commit_confirm` which should default to False). For such a fundamental change of behavior, just having a timeout is not a clear enough indicator that you are changing the behavior).

Reply:
it made sense to me implicitly also, but if people feel that explicit > implicit here I don't have a strong opinion.  If we stay with the implicit route I will expand the `confirm_timeout` docstring to better explain.

Reply:
Okay, yeah we definitely need `commit_confirm` argument also. Some of this is historical i.e. NAPALM `commit_config` didn't have this commit confirm functionality historically. Also some of it is platform specific i.e. on a set of platforms people won't be familiar with it. So NAPALM users will need to toggle commit_confirm=True to use the feature (and optionally could change the timeout).

Reply:
done